org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -119,6 +128,49 @@ protected void importProjects(Collection<IPath> rootPaths, IProgressMonitor moni
 		}
 	}
 
+	protected void importProjectsFromConfigurationFiles(Collection<IPath> rootPaths, Collection<IPath> projectConfigurations, IProgressMonitor monitor) throws OperationCanceledException, CoreException {
+		SubMonitor subMonitor = SubMonitor.convert(monitor, rootPaths.size() * 100);
+		for (IPath rootPath : rootPaths) {
+			File rootFolder = rootPath.toFile();
+			String configurationName = null;
+			for (IProjectImporter importer : importers()) {
+				if (importer instanceof GradleProjectImporter) {
+					configurationName = ".gradle";
+				} else if (importer instanceof MavenProjectImporter) {
+					configurationName = "pom.xml";
+				} else if (importer instanceof EclipseProjectImporter) {
+					configurationName = ".project";
+				} else {
```

Comment:
Using a hardcode file suffix to match the importer is not scalable. I would prefer to add a new ProjectImporter method `boolean applies(Collection<IPath> buildFiles, IProgressMonitor monitor)` instead of `void initialize(File rootFolder, Collection<java.nio.file.Path> directories)`. The new method will match the specified configuration file paths and the update the `directories` variable in ProjectImporter.

Reply:
Yes, that sounds better